src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdlib.h>
+#include "pal_locale_internal.h"
+#include "pal_collation.h"
+
+#import <Foundation/Foundation.h>
+
+#if defined(TARGET_OSX) || defined(TARGET_MACCATALYST) || defined(TARGET_IOS) || defined(TARGET_TVOS)
+
+// Enum that corresponds to C# CompareOptions
+typedef enum
+{
+    None = 0,
+    IgnoreCase = 1,
+    IgnoreNonSpace = 2,
+    IgnoreWidth = 16,
+    StringSort = 536870912,
+    
+} CompareOptions;
+
+static NSStringCompareOptions ConvertFromCompareOptionsToNSStringCompareOptions(int32_t comparisonOptions)
+{
+    switch(comparisonOptions)
+    {
+        case None:
+        case StringSort:
+            return NSLiteralSearch;
+        case IgnoreCase:
+            return NSCaseInsensitiveSearch;
+        case IgnoreNonSpace:
+            return NSDiacriticInsensitiveSearch;
+        case (IgnoreNonSpace | IgnoreCase):
+            return NSCaseInsensitiveSearch | NSDiacriticInsensitiveSearch;
+        case IgnoreWidth:
+            return NSWidthInsensitiveSearch;
+        case (IgnoreWidth | IgnoreCase):
+            return NSWidthInsensitiveSearch | NSCaseInsensitiveSearch;
+        case (IgnoreWidth | IgnoreNonSpace):
+            return NSWidthInsensitiveSearch | NSDiacriticInsensitiveSearch;
+        case (IgnoreWidth | IgnoreNonSpace | IgnoreCase):
+            return NSWidthInsensitiveSearch | NSDiacriticInsensitiveSearch | NSCaseInsensitiveSearch;
+        default:
+            return 0;
```

Comment:
Nit: If we are able to add options in the future, this format is `2^n` cases, and can quickly be cluttered compared to a direct mapping of NSStringCompareOptions for each bit that we see in the compareOptions.

Can `StringSort` be used in conjunction wtih other options? It didn't seem like there was a limitation like `Ordinal` in the docs.

Can `NSLiteralSearch` be combined with other options?
```
int32_t supportedOptions = IgnoreCase | IgnoreNonSpace | IgnoreWidth | StringSort;
NSStringCompareOptions options = 0;

if ((comparisonOptions | supportedOptions) != supportedOptions)
    return options;

if (comparisonOptions | IgnoreCase)
    options = options | NSCaseInsensitiveSearch;
    
if (comparisonOptions | IgnoreNonSpace)
    options = options | NSDiacriticInsensitiveSearch;

if (comparisonOptions | IgnoreWidth)
    options = options | NSWidthInsensitiveSearch;

if (options == 0 || comparisonOptions | StringSort)
    options = NSLiteralSearch; // Or options |= NSLiteralSearch; if we are allowing it to be combined?

return options;
```

I wonder if passing 0 as the NSStringCompareOptions still is considered NSLiteralSearch as noted in https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Strings/Articles/SearchingStrings.html#//apple_ref/doc/uid/20000149-CJBBGBAI

Reply:
1. Regarding `StringSort` current behaviour is that it will end up like None check this line https://github.com/dotnet/runtime/blob/main/src/native/libs/System.Globalization.Native/pal_collation.c#L679
I will also take this line in my code, so if `StringSort` will be used with other option it will be ignored.
3. Re: `Can NSLiteralSearch be combined with other options?` This is the case when user will pass `CompareOptions.None | CompareOptions.[SomeOtherOption]` yes? as `CompareOptions.None` is 0 it will be ignored at the end.
4. The code you mentioned will not work correct. For example `CompareOptions.IgnoreCase` is passed to the function. At the end options will be equal to `NSWidthInsensitiveSearch` as all if checks would be true.
If there will be added new option in `CompareOptions` there will be need first to find correct mapped option in `NSStringCompareOptions`, which I don't think is possible to implement with `|` operator.
5. Re:  `I wonder if passing 0 as the NSStringCompareOptions still is considered NSLiteralSearch as noted in` I couldn't find where it is mentioned that `NSLiteralSearch` is considered `0` in the link you mentioned. From here it is `2` https://developer.apple.com/documentation/foundation/nsstringcompareoptions/nsliteralsearch

Reply:
2. I was wondering what the behaviour should be if CompareOptions.StringSort | CompareOptions.IgnoreCase (0x20000001) were passed. 
1. Ah I see, so we will only check the lower 5 bits, do non alphanumeric characters still come before alphanumeric characters for `CompareOptions.IgnoreCase`? (wondering if there is a distinction between `NSLiteralSearch | NSCaseInsensitiveSearch` and `NSCaseInsensitiveSearch` for osx)
3. Ah my mistake, I should've put the `&` instead, that was it will only match if the correct bit is set.
`if (comparisonOptions & IgnoreCase)`
4. There was a `Search and comparison are currently performed as if the NSLiteralSearch option were specified.` blurb, but as it is the archive, not sure if something has changed since. It wasn't that NSLiteralSearch is considered 0, but rather that if 0 or no options are passed, I think it defaulted to NSLiteralSearch from the archive.